src/vscode-dts/vscode.d.ts

```diff
@@ -19896,19 +19903,23 @@ declare module 'vscode' {
 	/**
 	 * A token that can be passed to {@link lm.invokeTool} when invoking a tool inside the context of handling a chat request.
 	 */
-	export type ChatParticipantToolToken = unknown;
+	export type ChatParticipantToolToken = never;
```

Comment:
Is there anything better than `never`? I tried a class with private ctor but without fake-discriminating types anything like `{}` can be assigned to it...

Reply:
We can do runtime validation of the argument?

Reply:
`never` seems like that's not the right type, since tools won't be able to assign anything to the `toolInvocationToken`

https://www.typescriptlang.org/play/?#code/JYOwLgpgTgZghgYwgAgJIDED2nkG8BQyRyAHgFzIgQBu0yAPsgK4gAmEMoErA3PgL7587BABs4UFAkwgAzmGQxsFDNj5LMAOhLIAvMgCMAJgDMPIA

I think `unknown` is probably most correct for an opaque token.

Reply:
But you can only assign it to one thing, the token that the participant was given, which is also typed as `never`, so you assign `never` to `never`. So it's a bit weird but actually good that you can't assign anything else to it.

Reply:
Oh, I see, that's meant to be passed back and returned in the result. 

`unique symbol` probably does what you want https://www.typescriptlang.org/play/?#code/FAEwpgxgNghgTmABAWwPYgK5SQcgG4DOE6YOiA3sItYmAB4AOqcALogJYB2LYcAZjAhIAkgDFUqClRoyEMEKk5QAnojoAuRBk7sAjhiQFlyAEaooAbmnUAvsDvB2yJq0QAqRDAKJCxcIj44VGREfCISHCtwaHgkYk4CNj4JAEZNXxIAOjEJK2B4xICJACZ08PBs8UkAXikZDSLUFMy6ABp7YHzFQuTUAGYyvzBKiURaynrNHAB3RTZZuABrHHa7IA

Reply:
That's promising, unfortunately it has the weird caveat that you can't say `unique symbol | undefined`. I think we would want an object with a unique property like you suggest but it just starts to get uglier and more confusing 😄 I'm going to leave it alone for now and merge this, might revisit next month. I think it would be fine to change this type after the fact, it shouldn't break any correct code.